src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -660,6 +659,15 @@ private BoundExpression CreateAndPopulateArray(BoundCollectionExpression node, A
                     var rewrittenSpreadExpression = VisitExpression(spreadExpression);
                     return _factory.Call(rewrittenSpreadExpression, listToArrayMethod.AsMember((NamedTypeSymbol)spreadExpression.Type!));
                 }
+                else if (IsConvertibleToObject(arrayType.ElementType)
+                    && _factory.WellKnownMethod(WellKnownMember.System_Linq_Enumerable__ToArray) is { } linqToArrayMethodGeneric)
+                {
+                    var linqToArrayMethod = linqToArrayMethodGeneric.Construct([arrayType.ElementTypeWithAnnotations]);
+                    if (ShouldUseAddRangeOrToListMethod(spreadExpression.Type!, linqToArrayMethod.Parameters[0].Type, spreadElement.EnumeratorInfoOpt?.GetEnumeratorInfo.Method))
```

Comment:
>ShouldUseAddRangeOrToListMethod [](http://example.com/codeflow?start=24&length=31)

Does this method need a more general name since this case is neither `AddRange()` nor `ToList()`?